sql/core/src/main/scala/org/apache/spark/sql/execution/stat/StatFunctions.scala

```diff
@@ -264,7 +264,6 @@ object StatFunctions extends Logging {
     }
 
     val selectedCols = ds.logicalPlan.output
-      .filter(a => a.dataType.isInstanceOf[NumericType] || a.dataType.isInstanceOf[StringType])
```

Comment:
Can we keep this whitelist style instead of allowing all, @Fokko ?

Reply:
Sure, I've just updated the list.

Reply:
Thank you, @Fokko . Could you update the PR title and description accordingly?

Reply:
I've updated both the PR and commit, please let me know if this works for you

Reply:
Shall we remove `Even something less common such as arrays are sortable:` and its example?